Fix/post statuses in query

adekbadek opened the pull request:
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes #38

A question – is checking for `edit_others_posts` capability a safe-enough check? 

For design review – the label which marks non-published posts: 

<img width="479" alt="image" src="https://user-images.githubusercontent.com/7383192/163977780-814a935b-1e4e-43ac-a9c5-d373844eee16.png">

### How to test the changes in this Pull Request:

1. Insert Homepage Posts and Carousel blocks on a page
2. Ensure you have some recent draft and scheduled posts 
3. For both, observe a new panel in block settings – "Post Statuses". Set different values for this attribute, and observe the post list is updated – non-published posts should be displayed with a label
4. Publish and visit the page as the logged-in editor user – observe the posts are shown as in the editor
5. Visit the page as a non-logged-in user – observe only published posts are displayed

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

dkoo reviewed (changes requested):
This works well as described, but I'm uncertain about the idea of introducing a block attribute to show or not show preview posts.

Would it be enough and easier to understand if we simply showed all draft, scheduled, and published posts with labels for each non-published status in the block editor? On the front-end side, we could show these non-published posts if the user is logged in, has the `edit_others_posts` capability, and there's a `preview=true` param on the URL (meaning the post is visited from the "Preview" functionality in the editor). 

Or, to make it even simpler, just always show all of the draft, scheduled and published posts to logged-in users with the right capabilities. In this case if an editor really wants to see a post "as it would appear to readers" they can view it in an incognito window.

At the very least, I would advocate for a simpler interface: instead of being able to select Publish, Draft, and/or Scheduled, just make it a single toggle (off by default) to "Show draft and scheduled posts" with the same explanation instead. Right now I can't really think of a good use case for not showing published posts since these will always be shown to readers, right?

> A question – is checking for `edit_others_posts` capability a safe-enough check?

If we want to further lock it down we could add a nonce verification, but I think checking the login state plus `edit_others_posts` is probably enough.

adekbadek commented:
Thank you for your input, but I think it's based on assumptions about publishing workflows that we cannot make. 

> Would it be enough and easier to understand if we simply showed all draft, scheduled, and published posts with labels for each non-published status in the block editor?

> to make it even simpler, just always show all of the draft, scheduled and published posts to logged-in users 

I imagine a WP user might:

- Have a ton of insignificant drafts, and a few scheduled posts. They want to see how the block would look like after the scheduled posts are published. 
- Have a few drafts, which are planned for publication this week, and a couple of posts scheduled further into the future. They want to see the block as it would appear at the end of the week (drafts only).
- Not use scheduling and have a ton of insignificant drafts, which they don't ever want to see in the block.

This PR would support all these workflows, while the proposition in the comment would impair all of these, assuming drafts and scheduled posts are always **few** and **desired** when previewing.

I'm all for simplification, but this is an non-core feature, and one which which uses established CMS concepts, so I don't think it's too complicated.  

> Right now I can't really think of a good use case for not showing published posts since these will always be shown to readers, right?

That's right – how about removing the "published" status from the UI, so user can only select what they want to see in addition to the published posts/

dkoo commented:
> Thank you for your input, but I think it's based on assumptions about publishing workflows that we cannot make.

That's true—I think this is a feature which might benefit from some user feedback across sites of various sizes. I can certainly imagine the scenarios you describe but can't really say how many sites will actually encounter them.

> That's right – how about removing the "published" status from the UI, so user can only select what they want to see in addition to the published posts

I'm okay with this approach for now if you just want to get the feature out there.

adekbadek commented:
After an internal discussion:

- changed labels to specific posts status instead of generic "preview"
- removed "published" option from the checkbox-selection

dkoo commented on src/blocks/homepage-articles/view.scss:
```diff
@@ -758,3 +758,17 @@ amp-script .wpnbha.has-more-button.is-error {
 /* stylelint-enable */
+
+// Also used by the Carousel block.
+.newspack-preview-label {
```
These styles aren't enqueued if a post only contains a Carousel block and not the Homepage Posts block.

adekbadek replied:
Thanks for catching that! Fixed in a392e6ac4d78b86ca9bb58c65f71b027227cae2e

dkoo reviewed (changes requested):
A minor style issue with Post Carousels, otherwise looking 👍
